packages/lodestar/test/unit/chain/factory/block/blockAssembly.test.ts

```diff
@@ -12,28 +12,30 @@ import { StatefulDagLMDGHOST } from "../../../../../../lodestar/src/chain/forkCh
 import { BeaconChain } from "../../../../../src/chain";
 import { generateEmptyBlock, generateEmptySignedBlock } from "../../../../utils/block";
 import { BlockRepository, DepositDataRootListRepository, StateRepository } from "../../../../../src/db/api/beacon/repositories";
+import {StubbedBeaconDb, StubbedChain} from "../../../../utils/stub";
+import { IBeaconDb } from "../../../../../src/db";
 
 describe("block assembly", function () {
 
   const sandbox = sinon.createSandbox();
 
-  let assembleBodyStub: any, chainStub: any, forkChoiceStub: any, stateTransitionStub: any, opPool: any, beaconDB: any, eth1: any;
+  let assembleBodyStub: any, chainStub: StubbedChain, forkChoiceStub: any, stateTransitionStub: any, opPool: any, beaconDB: StubbedBeaconDb, eth1: any;
 
   beforeEach(() => {
     assembleBodyStub = sandbox.stub(blockBodyAssembly, "assembleBody");
     stateTransitionStub = sandbox.stub(blockTransitions, "stateTransition");
 
 
     forkChoiceStub = sandbox.createStubInstance(StatefulDagLMDGHOST);
-    chainStub = sandbox.createStubInstance(BeaconChain);
+    chainStub = sandbox.createStubInstance(BeaconChain) as unknown as StubbedChain;
     chainStub.forkChoice = forkChoiceStub;
 
     opPool = sandbox.createStubInstance(OpPool);
     beaconDB = {
       block: sandbox.createStubInstance(BlockRepository),
       state: sandbox.createStubInstance(StateRepository),
       depositDataRootList: sandbox.createStubInstance(DepositDataRootListRepository)
-    };
+    } as StubbedBeaconDb;
```

Comment:
It will probably be useful if we had:
```typescript
export class MockDb extends IBeaconDb {
   block: SinnonStubbedInstance<IBlockRepository>;
   ...

   constructor(sinon) {
      block = sinon.createStubbedInstance(BlockRepository);
     ...
   }
}
```

Reply:
Agreed, would definitely be more robust. Going to make an issue for this.

Reply:
#859